src/parser/parse_tree.h

```diff
@@ -114,6 +114,7 @@ struct json_t;
 
 #define PT_IS_QUERY_NODE_TYPE(x) \
     (  (x) == PT_SELECT     || (x) == PT_UNION \
+    || (x) == PT_DBLINK_TABLE \
```

Comment:
There are so many places where PT_IS_QUERY_NODE_TYPE() is used that it looks like you need to be careful.

How about modifying it like the one below where it needs to be processed?
`if (PT_IS_QUERY_NODE_TYPE (node->node_type) || node->node_type == PT_DBLINK_TABLE)`

In a similar form, there is a part expressed as below in the go_optimize_queries() function.
`if (PT_IS_QUERY_NODE_TYPE (node->node_type) || node->node_type == PT_UPDATE || node->node_type == PT_DELETE)`

Reply:
Ok, I agree.